homeassistant/helpers/entity_platform.py

```diff
@@ -339,7 +340,7 @@ async def _async_setup_platform(
         )
         with async_start_setup(hass, [full_name]):
             try:
-                task = async_create_setup_task()
+                task = create_eager_task(async_create_setup_task())
```

Comment:
Should we be able to call `task.done()` to avoid awaiting shield? Or would shield not suspend when task is already done?

Reply:
shield is always going to wrap a coro in a task to schedule it afaik

Reply:
https://github.com/python/cpython/blob/c562f5a1e2275e8533313a52ab3b0831b0e1f5cb/Lib/asyncio/tasks.py#L884